### DISCUSSION TITLE
v2.3.000

### DISCUSSION THREAD

**Discussion by Medard22** [Medard22/Dolphin-MMJR2-VBI#579] [Announcements]
### **Changelog for v2.3.000 (from v2.2.110)**

- Extensive code refactors including a major overhaul of the graphics pipeline
- Graphics pipeline code revision - removed obsolete MMJR2-specific hacks, that are no longer needed
- Revised and improved Asset Loader
- Updated external components like SDL, Cubeb, mGBA, fmt, etc.
- Updated NDK to v25.2.9519653
- Updated Android libraries, now targeting Java 17
- Many JIT optimizations
- Minor updates to the Vulkan backend
- Optimized Vertex Loader
- Updated GameINI database
- Fixed Fast Forward hotkey
- Added support for WiiConnect24 via WiiLink
- Added support for Color Space Correction
- Leveraged LZ4 algorithm for faster savestate decompression
- Updated and fixed multiple instances of partially synced code across the project

### **Missing features - these features from Dolphin Official are not implemented**

- Android Input Overhaul and subsequent changes to Input and TAS handling
- Custom GPU driver support
- Skylanders portal and Infinity base support
- Gamma correction

**This codebase is roughly equivalent to the Official Dolphin dev build 5.0-20223 (Oct 15, 2023).**

### General information
This release includes over 7 months' worth of code changes ported from Dolphin Official since v2.2.110, and more than 3 months since the last beta. I’ve tested it extensively, but as always, your mileage may vary. 🤷‍♂️ I upgraded my RP3+ to a Retroid Pocket 4 Pro and performed most of the testing on this device.

Android developers willing to help are always welcome! 

### Caution

- Save States from v2.2.110, v2.3.000-rc1, or v2.3.000-rc2 are NOT compatible with this release. Be sure to have in-game saves before updating!
- Due to the graphics pipeline changes, this build bumps the shader cache UID version. Shaders will need to be rebuilt from scratch, so expect some initial shader compilation stutter.

### Known Issues

The UI in the Cheats section is broken when you try to delete multiple items in a row. This bug is present to this day in Dolphin Official. Just delete a few items, exit the cheats window, and return back.

As always, have fun and happy gaming ✌️

<hr /><em>This discussion was created from the release <a href='https://github.com/Medard22/Dolphin-MMJR2-VBI/releases/tag/'>v2.3.000</a>.</em>

**Comment by MarinaDaianaRosa** [comment]
Do you ever plan on adding the Skylanders portal?

**Reply by Medard22** [reply]
The support for Skylanders portal and Infinity base is present in the core emulator and believe me I would love to add the feature if I could. The problem lies with the Android-specific code. MMJR2 uses an older version of the Android UI. For bigger features like Skylanders support, you need to build a piece of UI from scratch, which is something I can't do right now. I know the code well enough to reuse parts of the UI when editing settings - adding simple features like on/off switches or sliders, but I don't know how to build something complex like a Skylanders UI. Since MMJR2 is built on an old codebase, I cannot use the Dolphin official code.
Custom GPU drivers are not available for the very same reason. As for the gamma curve, I couldn't figure out how to pass expected float values to the core because all UI elements (sliders) operate with integers. The whole Android part could use an overhaul. I am aware of the issues it has, but with my current knowledge, there isn't much I can do here. If you know of any developer with experience in Android programming who is willing to help, feel free to send them my way.

**Reply by MarinaDaianaRosa** [reply]
I really wish I could recommend someone who could help, but unfortunately I don't know anyone who can, sorry.

Even though I can't help, I'm glad to know that you would be willing to add the portal. I wish you the best of luck with the project, your emulator is the best for Android

**Comment by nobidebian** [comment]
I have a problem when running bully scholarship edition it runs 0 fps, but on Official Dolphin it works perfectly fine without any settings configured.

**Reply by Medard22** [reply]
Hi. Wii version of Bully is a hot mess. The wiimote intro screen is extremely slow. That is a known bug and it happens in the Dolphin Official too. It can take up to 5-10 minutes to get past it. The gameplay after that should be ok-ish, just use savestate to bypass the intro screen, or even better play the ps2 version.

**Comment by ahmedtalha31** [comment]
Please use in the next update the last version of the original as it gives more fps but has controller problem

**Reply by Medard22** [reply]
You are missing the point of this fork. You better check Dolphin bug tracker and report there - [https://bugs.dolphin-emu.org/issues](https://bugs.dolphin-emu.org/issues)
